src/tutorials/Data-Structures/02.md

```diff
@@ -5,17 +5,15 @@ Before we dig into how sharing works on the decentralized web, let's take a minu
 
 URLs (Uniform Resource Locators) are the primary addresses we give each other for data on the centralized web (you know, that plain old web we're all used to). They make it possible for us to create links and connect data on the web, so they serve a valuable purpose. (The web would be pretty lousy without links!) However, <strong>URLs are based on the <em>location</em> where data is stored</strong>, not on the contents of the resource stored there. We call this <strong>location addressing</strong>, and it presents us with some problems.
 
-Most of us have a lot of experience with URLs, and we make some assumptions about them based on our experience. When we see  `https://www.puppies.com/beagle.jpg`, for example, we'll probably guess from the file name and extension that
-the data stored at that location is an image (in JPEG format) of a beagle, but we can't verify this from the URL alone. There may very well be a photo of a chihuahua hiding at `beagle.jpg`, or even worse, an adorable kitty!
+Most of us have a lot of experience with URLs, and we make some assumptions about them based on our experience. When we see  `https://www.puppies.com/beagle.jpg`, for example, we'll probably guess from the file name and extension that the data stored at that location is an image of a beagle (in JPEG format), but we can't verify this from the URL alone. There may very well be a photo of a chihuahua hiding at `beagle.jpg`, or even worse, an adorable kitty!
 
-Through the domain name, URLs indicate the *authority* we should ask for the data. Even though the web is already decentralized, in that anyone can link to anyone else, the location-based references we use for those links require that the data itself be centralized so that we can get it from an authority. We make assumptions about these authorities or domains as well. For example, we'd assume that a file hosted at `puppies.com` is safer to open than one hosted at `evilhacker.com`, but we can't be sure of it.
+Through the domain name, URLs indicate which *authority* we should go to for the data. Even though the web is already decentralized in a sense (since anyone can link to anyone else), the links referencing the data are location-based. So the data itself must be centralized on an authority in order for us to find it. We also make assumptions about these authorities (or domains), just as we do with file names. For example, we might assume that a file hosted at `puppies.com` is safer to open than one hosted at `evilhacker.com`, but we can't be sure of it.
 
 All of this uncertainty also works in reverse. If we see a picture of an adorable puppy and are told it's stored on the web, there's no way for us to guess the URL that would lead us to the image. We can neither determine the domain, which tells us who's hosting it, nor the filename.
```

Comment:
I'm not sure what the “All of this uncertainty works in reverse…” means.

Reply:
@ericronne Does this make sense to you? "Ultimately, the contents of a file hosted on the centralized web have no direct relationship with their location-based addresses."

Reply:
Way more clear to me, thx